view/frontend/web/js/bolt-api-driven-checkout.js

```diff
@@ -1077,9 +1102,39 @@ define([
                 this.hintsBarrier.resolve({"hints": this.boltCartHints})
                 BoltCheckoutApiDriven.isPromisesResolved = true;
             }
+
+            if (this.boltSubscriptionsEnabled) {
+                this.boltSetUpSubscriptionFrequencySelector();
+            }
         }
     }
 
+    $(document).on('ajax:addToCart', async function (event, data) {
+        if (!BoltCheckoutApiDriven.boltSubscriptionsEnabled) {
+            return;
+        }
+
+        const sleep = (millisec) => {
+            return new Promise(resolve => {
+                setTimeout(() => { resolve('') }, millisec);
+            })
+        }
+
+        // When an item is added to an empty cart, the quoteMaskedId takes a moment to get populated, so wait for that
+        const maxAttempts = 3;
+        for (let i = 0; i < maxAttempts; i++) {
+            if (!BoltCheckoutApiDriven.quoteMaskedId) {
+                await sleep(1000);
```

Comment:
Could we remove this sleep code?
I still think that moving wait logic to storm is the best approach, so we can have two API calls:
`window.BoltCheckout.persistSubscriptionFrequencySelection(quoteMaskedId) where quoteMaskedId may be null
`and
`window.BoltCheckout.setCartToken`

but as a simple middleground we can set local flag
```
if (BoltCheckoutApiDriven.quoteMaskedId) {
            window.BoltCheckout.persistSubscriptionFrequencySelection(BoltCheckoutApiDriven.quoteMaskedId);
} else {
        this.shouldCallPersistSubscriptionFrequencySelection = true
}
```
and after line
`BoltCheckoutApiDriven.quoteMaskedId = magentoCart.quoteMaskedId;`
add
```
if this.shouldCallPersistSubscriptionFrequencySelection {
 BoltCheckoutApiDriven.quoteMaskedId = magentoCart.quoteMaskedId;
 this.shouldCallPersistSubscriptionFrequencySelection = false
}
```